lib/main.js

```diff
@@ -542,53 +544,30 @@ module.exports = class AssetPipeClient {
     }
 
     middleware() {
-        const devMiddleware = new AssetDevMiddleware(
-            [this.assets.js].filter(Boolean),
-            [this.assets.css].filter(Boolean)
-        );
-
-        this.transforms.forEach(entry => {
-            devMiddleware.transform(entry.transform, entry.options);
-        });
+        if (this.development) {
+            const devMiddleware = new AssetDevMiddleware(
+                [this.assets.js].filter(Boolean),
+                [this.assets.css].filter(Boolean)
+            );
 
-        this.plugins.forEach(entry => {
-            devMiddleware.plugin(entry.plugin, entry.options);
-        });
+            this.transforms.forEach(entry => {
+                devMiddleware.transform(entry.transform, entry.options);
+            });
 
-        const middlewares = [(req, res, next) => next()];
+            this.plugins.forEach(entry => {
+                devMiddleware.plugin(entry.plugin, entry.options);
+            });
 
-        if (this.development) {
             // middleware to serve up development assets at /js and /css
-            middlewares.push(devMiddleware.router());
+            return devMiddleware.router();
         } else {
-            // publish js assets
-            if (this.assets.js) {
-                const promise = this.publishAssets(
-                    this.tag,
-                    this.assets.js
-                ).then(({ id }) => (this.hashes.js = id));
-
-                this.publishPromises.push(promise);
-            }
-
-            // publish css assets
-            if (this.assets.css) {
-                const promise = this.publishAssets(
-                    this.tag,
-                    this.assets.css
-                ).then(({ id }) => (this.hashes.css = id));
-
-                this.publishPromises.push(promise);
-            }
-
             // middleware to ensure that assets are uploaded
-            middlewares.push(async (req, res, next) => {
-                await Promise.all(this.publishPromises);
-                next();
-            });
+            return (req, res, next) => {
+                this.ready()
+                    .then(() => next())
```

Comment:
Do you need to do `() => ... )`here? Can you just pass on `next` as done on catch?

Reply:
Because next follows the err back pattern, if I just pass next to then directly then any resolved value will go straight into the first (error) argument slot of the next function so next will think there is an error when there isn’t.